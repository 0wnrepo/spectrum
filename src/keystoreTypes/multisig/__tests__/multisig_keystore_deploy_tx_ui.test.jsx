import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import MultisigKeystoreDeployTxUi from '../multisig_keystore_deploy_tx_ui';

const address = {
  keystore: {
    type: {
      color: 'black',
      icon: 'testIcon',
    },
  },
  name: 'test network',
  address: '0x213452',
  color: 'blue',
};

const network = {
  explorerAddressPrefix: 'testPrefix',
  name: 'test network',
};

const ui = {
  required: 2,
  count: 5,
};

describe('<MultisigKeystoreDeployTxUi />', () => {
  test('renders correctly', () => {
    const component = shallow(
      <MultisigKeystoreDeployTxUi
        address={address}
        network={network}
        ui={ui}
      />,
    );

    expect(toJson(component)).toMatchSnapshot();
  });
});
