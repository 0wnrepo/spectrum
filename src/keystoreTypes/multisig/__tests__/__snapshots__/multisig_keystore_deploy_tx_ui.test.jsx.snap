// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MultisigKeystoreDeployTxUi /> renders correctly 1`] = `
<TxVisualisation
  items={
    Array [
      Object {
        "color": "black",
        "data": "0x213452",
        "dataLink": "testPrefix0x213452",
        "header": "Sender",
        "icon": "testIcon",
        "subheader": "test network",
      },
      Object {
        "color": "blue",
        "data": "test network network",
        "header": "Deploy",
        "icon": "upload",
        "subheader": "Create new Contract",
      },
      Object {
        "color": "orange",
        "data": "2 approvals required",
        "header": "Multisig Wallet",
        "icon": "file text outline",
        "subheader": "5 owners",
      },
    ]
  }
/>
`;
